Code/C++/avl_tree.cpp

```diff
@@ -0,0 +1,429 @@
+#include <bits/stdc++.h>
```

Comment:
add description of the algorithm at the top.

Reply:
done sir, please review